test/ServiceBus.Tests/TestStreamProviders/StreamPerPartitionEventHubStreamProvider.cs

```diff
@@ -19,27 +22,36 @@ namespace ServiceBus.Tests.TestStreamProviders.EventHub
     [Collection(TestEnvironmentFixture.DefaultCollection)]
     public class StreamPerPartitionEventHubStreamProvider : PersistentStreamProvider<StreamPerPartitionEventHubStreamProvider.AdapterFactory>
     {
-        public class AdapterFactory : EventHubAdapterFactory
+        private class CacheFactory : IEventHubQueueCacheFactory
         {
-            private TimePurgePredicate timePurgePredicate = null;
+            private readonly EventHubStreamProviderSettings adapterSettings;
+            private readonly SerializationManager serializationManager;
+            private readonly Logger log;
 
-            public AdapterFactory()
+            public CacheFactory(EventHubStreamProviderSettings adapterSettings, SerializationManager serializationManager, Logger log)
             {
-                CacheFactory = CreateQueueCache;
+                this.adapterSettings = adapterSettings;
+                this.serializationManager = serializationManager;
+                this.log = log;
```

Comment:
why are we passing in the log at cach factory creation time and using the same log per cache, even though a log is provided for each cach in the create cache call?